CMakeLists.txt

```diff
@@ -795,31 +839,17 @@ if (HDF5_ENABLE_SUBFILING_VFD)
   if (NOT HDF5_ENABLE_PARALLEL)
     message (FATAL_ERROR "Subfiling VFD requires a parallel HDF5 build")
   else ()
-    # Check for MPI_Comm_split_type
-    CHECK_SYMBOL_EXISTS (MPI_Comm_split_type "mpi.h" H5_HAVE_MPI_Comm_split_type)
+    # Make sure we found MPI_Comm_split_type previously
     if (NOT H5_HAVE_MPI_Comm_split_type)
       message (FATAL_ERROR "Subfiling VFD requires MPI-3 support for MPI_Comm_split_type")
     endif ()
   endif ()
 
-  if (NOT DEFINED Threads_FOUND)
-    set (THREADS_PREFER_PTHREAD_FLAG ON)
-    find_package (Threads REQUIRED)
+  # Subfiling requires thread operations
+  if (NOT Threads_FOUND)
+    message (FATAL_ERROR "Subfiling requires thread operations support")
   endif ()
 
-  # For now, make sure we're using pthreads. Once Subfiling can be
-  # supported on Windows, we should allow Win32 threads as well
-  if (NOT ${Threads_FOUND} OR NOT ${CMAKE_USE_PTHREADS_INIT})
```

Comment:
Without the threading changes in develop, I'm not sure that changing the CMake here around the Subfiling VFD is a good idea.

Reply:
Will fix